src/easydmp/plan/models.py

```diff
@@ -166,8 +172,42 @@ def clone(self, plan):
         new.save()
         return new
 
+    def validate(self, timestamp: datetime = None, _data: Dict = None) -> bool:
+        """
+        Validates the answers and persists the validity state on this as well as on related Answers.
+        (If _data is not given, use data on this.) It is legal to pass a superset of the data required to validate
+        the questions belonging to this. For example, pass the entire data dict for a plan.
+
+        Returns validity.
+        """
+        # TODO We plan to stop passing _data when the data is already being stored on the AnswerSet
+        data = _data or self.data
+        valids, invalids = self.section.find_validity_of_questions(data)
+        self.set_validity_of_answers(valids, invalids)
+        self.last_validated = timestamp or tznow()
+        self.valid = not invalids
+        self.save()
+        return self.valid
+
+    def set_validity_of_answers(self, valids: Set[int], invalids: Set[int]) -> None:
+        """
+        Update the corresponding Answers of this with validities/invalidities given by the passed Question pks
+        """
+        for answer in self.answers.all():
+            if answer.question.pk in valids:
+                answer.valid = True
+            elif answer.question.pk in invalids:
+                answer.valid = False
+            else:
+                raise ValueError('No question for answer {} in validation'.format(answer.pk))
+            answer.save()
+
```

Comment:
I agree with codecov. We need tests! With a template export, we can create an template, export it and import it in a test instead of hand-coding a structure.